### PR TITLE
Add new migration instructions feature flag

### PR DESCRIPTION
**Pull request by renatho**
Related to #91773

## Proposed Changes

* It adds feature flag for the new migration instructions step.
* It also adds a new component where we can develop the new step.

## Why are these changes being made?

* We're preparing the codebase to start the development of the new migration instructions step.

## Testing Instructions

* Make sure to run `yarn start` again to the new feature flags have effect.
* Test the migration flow in the 3 different flows and make sure it's loading a new component instead of the current migration instructions.
  * `/start`
  * `/setup/migration-signup`
  * `/setup/hosted-site-migration?from={FROM_URL}` (It starts in https://wordpress.com/move)
* Now update the feature flag to `false` also in development, and run the `yarn start` again.
* Run the flows again, and make sure you see the original behavior.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: config/development.json [hunk 1]
```diff
@@ -155,6 +155,7 @@
 		"onboarding/import-light": false,
 		"onboarding/import-redirect-to-themes": true,
 		"onboarding/interval-dropdown": true,
+		"onboarding/new-migration-instructions-step": true,
```
Comment (donnapep) [inline config/development.json]: Not a big deal, but I'm not sure that this is technically an onboarding flow. Could we put it under `migration-flow/` as [Serenity did previously](https://github.com/Automattic/wp-calypso/pull/90663/files#diff-811c6e3fd087924c069c3a6a662ecf053f21649dabd23857e6d6c431e0d7a17fL142)?

**Review by donnapep — APPROVED** [review]
Works! Left a small comment about naming.

File: config/development.json (hunk 1 above)
Comment (renatho) [reply config/development.json]: Of course! Changed in https://github.com/Automattic/wp-calypso/pull/91820/commits/53409aa864db8bfbd044113850f9ce7604c783c9.

**Comment by renatho** [comment]
I'm going ahead with the deploy since the E2E errors are not related to this PR and are happening in all PRs.

**Pull request merged**